src/nvim/fileio.c

```diff
@@ -5755,7 +5755,7 @@ static int do_autocmd_event(event_T event, char_u *pat, int nested, char_u *cmd,
      */
     brace_level = 0;
     for (endpat = pat; *endpat && (*endpat != ',' || brace_level
-                                   || endpat[-1] == '\\'); ++endpat) {
+                 || (endpat > pat && endpat[-1] == '\\')); ++endpat) {
```

Comment:
Instead of checking for a leading ',' after the loop, we can do it before the loop. With that the first loop iteration will never check `endpat[-1]` and for all other iteration the check is okay.

```diff
@@ -5753,16 +5753,19 @@ static int do_autocmd_event(event_T event, char_u *pat, int nested, char_u *cmd,
      * Find end of the pattern.
      * Watch out for a comma in braces, like "*.\{obj,o\}".
      */
+    endpat = pat;
+    // ignore single comma
+    if (*endpat == ',') {
+      continue;
+    }
     brace_level = 0;
-    for (endpat = pat; *endpat && (*endpat != ',' || brace_level
-                 || (endpat > pat && endpat[-1] == '\\')); ++endpat) {
+    for (; *endpat && (*endpat != ',' || brace_level || endpat[-1] == '\\');
+         ++endpat) {
       if (*endpat == '{')
         brace_level++;
       else if (*endpat == '}')
         brace_level--;
     }
-    if (pat == endpat)                  /* ignore single comma */
-      continue;
     patlen = (int)(endpat - pat);
 
     /*
```